sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/schema/BeamSqlRecord.java

```diff
@@ -17,24 +17,25 @@
  */
 package org.apache.beam.sdk.extensions.sql.schema;
 
-import com.google.auto.value.AutoValue;
-import java.io.Serializable;
 import java.util.List;
+import org.apache.beam.sdk.values.BeamRecord;
+import org.apache.beam.sdk.values.PCollection;
 
 /**
- * Field type information in {@link BeamSqlRow}.
- *
+ * {@link BeamSqlRecord} represents one row element in a {@link PCollection},
+ * with type provider {@link BeamSqlRecordTypeProvider}.
  */
-@AutoValue
-public abstract class BeamSqlRowType implements Serializable {
-  public abstract List<String> getFieldsName();
-  public abstract List<Integer> getFieldsType();
+public class BeamSqlRecord extends BeamRecord {
```

Comment:
Let's merge the discussion of `BeamSqlRecord` and `BeamSqlRecordCoder`.

A `BeamSqlRecordHelper` class would be introduced, which has two methods
```
public static BeamSqlRecordType getSqlRecordType(BeamRecord);
public static BeamRecordCoder getSqlRecordCoder(BeamSqlRecordType);
```
1. I don't like the CAST mixing in code, as we know a CAST is must, let's centralize it for better document;
2. As mentioned above, I'll prepare some inner Coders to cover SQL types. Would create a separated task to expose in sdk/core;

Thanks for the naming suggestion, I would do it after we clear all the questions. --A simple rename would cause hundreds of lines impacted.